core/src/main/scala/org/apache/spark/util/Utils.scala

```diff
@@ -315,6 +315,37 @@ private[spark] object Utils extends Logging {
     dir.getCanonicalFile
   }
 
+  /**
+   * Create a directory that is writable by the group.
+   * Grant 770 permission so the shuffle server can create subdirs/files within the merge folder.
+   */
+  def createDirWith770(dirToCreate: File): Unit = {
```

Comment:
Shall we add the permission as a parameter and make this function more general considering it's a util function?